xarray/core/pdcompat.py

```diff
@@ -73,13 +75,29 @@ def __repr__(self) -> str:
 NoDefault = Literal[_NoDefault.no_default]  # For typing following pandas
 
 
-def nanosecond_precision_timestamp(*args, **kwargs) -> pd.Timestamp:
-    """Return a nanosecond-precision Timestamp object.
+def _timestamp_as_unit(date: pd.Timestamp, unit: str) -> pd.Timestamp:
+    # compatibility function for pandas issue
+    # where "as_unit" is not defined for pandas.Timestamp
+    # in pandas versions < 2.2
+    # can be removed minimum pandas version is >= 2.2
+    unit = cast(PDDatetimeUnitOptions, unit)
+    if hasattr(date, "as_unit"):
+        date = date.as_unit(unit)
+    elif hasattr(date, "_as_unit"):
+        date = date._as_unit(unit)
+    return date
+
+
+def default_precision_timestamp(*args, **kwargs) -> pd.Timestamp:
+    """Return a Timestamp object with the default precision.
 
-    Note this function should no longer be needed after addressing GitHub issue
-    #7493.
+    Xarray default is "ns". This can be overridden by setting
+    set_options(time_resolution="us") or any other resolution
+    of {"s", "ms", "us", "ns"}.
     """
-    if Version(pd.__version__) >= Version("2.0.0"):
-        return pd.Timestamp(*args, **kwargs).as_unit("ns")
-    else:
-        return pd.Timestamp(*args, **kwargs)
+    dt = pd.Timestamp(*args, **kwargs)
+    units = ["s", "ms", "us", "ns"]
+    default = _get_datetime_resolution()
+    if units.index(default) > units.index(dt.unit):
+        dt = _timestamp_as_unit(dt, default)
```

Comment:
This is a little surprising. From the docstring of this function, I would expect it to always convert the units -- or perhaps raise an error is that isn't possible.

Reply:
OK, maybe the naming of the function/docstring is not really appropriate. This should only convert to higher resolutions, if needed. But not the other way round, as this might mean loss of resolution. I'll look into this, thanks!